turborepo-tests/integration/tests/clone/basic-monorepo.t

```diff
@@ -0,0 +1,19 @@
+Setup
+  $ . ${TESTDIR}/../../../helpers/setup_integration_test.sh
+
+  $ git config uploadpack.allowFilter true
+  $ cd ..
+Make sure we allow partial clones
+
+Do a blobless clone
+  $ ${TURBO} clone file://$(pwd)/basic-monorepo.t basic-monorepo-blobless --local
```

Comment:
I'm not convinced we're actually doing a real blobless clone here. If anybody has ideas for better testing that doesn't involve hitting a remote git repo (unless we really have to ig), I'm open to ideas.

Reply:
Like the config claims it's blobless but my suspicion is that git only does a blobless clone if you really need it. For trivial repos it just doesn't make sense. I tried adding a test that adds a file, then deletes in in a second commit, and then does a blobless clone, with the idea that the file should *not* be included in the clone since it's not a blob that is reachable from HEAD. But the file was indeed cloned, so who knows.

Reply:
If you're not convinced this test is actually testing the behavior we want than we shouldn't include it. Only suggestion for testing would be to set up a local git server and clone from that, but that feels a bit like overkill.

Reply:
I would prefer not to do that, but if we need a specific response from a git server we probably either need to hit GitHub or somehow run a similar server locally. Since we're still exploring this I think it's fine to test it out where we want to use it instead of investing into a testing strategy. I do still stand that if you don't think this test is testing the behavior, we should remove it.

Reply:
So I think this test is still worth doing because it's testing that we have indeed set up the repo to be a treeless/blobless clone. From there, it's up to the server to respect this config, but we can't control that.